deploy/helm/templates/asserts.yaml

```diff
@@ -8,4 +8,12 @@
 
 {{- if mustRegexMatch "^[\\s]+$" .Values.cluster.uid -}}
 {{ fail "If specified, the custom cluster UID should be a valid string." }}
+{{- end -}}
+
+{{- if empty .Capabilities.HelmVersion.Version -}}
+{{ fail "This version of Helm is not supported. Please use 3.9.0 or newer." }}
+{{- end -}}
+
+{{- if eq ((semver "3.9.0").Compare (semver .Capabilities.HelmVersion.Version)) 1 -}}
```

Comment:
Technically speaking, this check is not needed, since right now, it only applies to 3.9.0-rc.1 (everything older is also covered by the check above). But I'd keep it to make the requirement explicitly stated here in the code.